### PR TITLE
Add Kinesis Firehose shortcut

### PR DESCRIPTION
**Pull request by danmactough**
Adds a new shortcut for creating a Kinesis Firehose that writes out to the specified S3 bucket. It can be a "direct put" firehose (with `firehose.putRecord()` or `firehose.putRecordBatch()`), or you can specify an input Kinesis Stream ARN.

### CODE REVIEW

**Comment by danmactough** [comment]
Thanks @rclark. I didn't want to over-engineer it before getting feedback; I appreciate your points.

I think the way we should model the firehose shortcuts is:

- A base class that implements the delivery stream, optionally configurable to read from a Kinesis Stream source; and
- A subclass for each destination configuration (S3, Elastic, Redshift, etc.)

So, I'd like to limit this PR to the base class + S3 destination.

Unlike the Lambda shortcut, the base firehose wouldn't be usable without the additional functionality of one of the subclasses (or your own custom subclass).

**Comment by rclark** [comment]
> Unlike the Lambda shortcut, the base firehose wouldn't be usable without the additional functionality

That's fine -- just don't have the cloudfriend module export the "abstract" class?

**Comment by danmactough** [comment]
> just don't have the cloudfriend module export the "abstract" class?

That's what I'm thinking, but the downside would be that you wouldn't be able to implement a custom subclass. I think that better fits the spirit of this project, though.

**Comment by danmactough** [comment]
@rclark Implemented something via 807b43c. Thoughts?

File: lib/shortcuts/s3-kinesis-firehose.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
+'use strict';
+const KinesisFirehoseBase = require('./kinesis-firehose-base');
+const assert = require('assert');
+
+'use strict';
+
+/**
+ * Creates a Kinesis Firehouse that can receive records by direct put or by consuming a Kinesis Stream
+ * and writes out to the specific S3 destination. Creates a Kinesis Firehouse delivery stream,
+ * sets up logging and creates a policy allowing records to be delivered to the delivery stream.
```
[the remark this answers is not in the document]
Comment (danmactough) [reply lib/shortcuts/s3-kinesis-firehose.js]: @rclark Added a freshness alarm via 09657e9. I chose to use `Average` as the statistic, which seems like a good default to me, but maybe you have an idea that this should be `Maximum` or `p9x`?

**Comment by danmactough** [comment]
> That's fine -- just don't have the cloudfriend module export the "abstract" class?

Yup

File: lib/shortcuts/s3-kinesis-firehose.js [hunk 2]
```diff
@@ -41,7 +53,24 @@ class S3KinesisFirehose extends KinesisFirehoseBase {
...
+      AlarmDescription = {
+        'Fn::Sub': [
+          'Freshness alarm for ${AWS::StackName}-${name} kinesis firehose in ${AWS::StackName} stack',
+          { name: LogicalName }
+        ]
+      },
+      AlarmActions = [],
+      Period = 60,
+      EvaluationPeriods = 1,
+      Statistic = 'Average',
```
Comment (rclark) [inline lib/shortcuts/s3-kinesis-firehose.js]: It should be on Max. For each time interval there is one data point per firehose "shard". The case you really want to catch here is one where shards are imbalanced, and a one shard is significantly more latent than another.

Comment (danmactough) [reply lib/shortcuts/s3-kinesis-firehose.js]: cool. done via 854201c

File: lib/shortcuts/s3-kinesis-firehose.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rclark) [reply lib/shortcuts/s3-kinesis-firehose.js]: Last thing: could you add a sentence or two here describing the alarm that is created?

**Comment by danmactough** [comment]
> Last thing: could you add a sentence or two here describing the alarm that is created?

Done via 9d2eebd

**Pull request merged**